WhitmanCSCapstone/ge-scheduling
# Add git hooks for linting and prettifying

I used Prettier, Lynt, Husky, and Lint-Staged using this guide recommended by Janet Davis:

https://blog.bitsrc.io/continuous-integration-with-prettier-eslint-c4f4e61b550b

I also added some good stuff.

To test:

- Get the latest changes
    - `git checkout lint`
    - `git pull`
- Try making a mess that will obviously cause an error. Add a mess of
special characters or something.
- `git commit` with no message, just to see if you can. This should
fail. If it succeeds, Lynt isn't working.
- Undo those changes and create a readability issue. Maybe remove some
line breaks.
- `git commit` with no message. Your default `$EDITOR` should open. Exit
the editor without writing a message to abort the commit.
- Check if the readability issues you introduced have been reverted.
    - If they have not, Prettier isn't working.

Resolves #10
